x/programs/rust/wasmlanche-sdk/src/state.rs

```diff
@@ -99,6 +99,10 @@ impl<'a, K: Key> State<'a, K> {
         let mut cache = self.cache.borrow_mut();
 
         let val_bytes = if let Some(val) = cache.get(&key) {
+            if val.is_empty() {
+                return Ok(None);
+            }
```

Comment:
🤔 we should probably make sure empty `Vec`s don't get inserted. Can you create an issue for that after you merge this?

Reply:
I updated the code so that it checks in the `store` function if the serialized data is empty, thanks for that suggestion!